specifications_dbus.md

```diff
@@ -0,0 +1,92 @@
+
+On subsequent startups, the connector should call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) with the *same* token, and store the new endpoint returned, if it has changed. The connector should also implement the [org.unifiedpush.Connector1.NewEndpoint](#orgunifiedpushconnector1newendpoint-string-string--nothing) method to receive notification about a changed endpoint from the distributor at any time.
+
+The application SHOULD provide an activatable D-Bus service, so that it can be activated by a new push message when it is not running. See [Appendix: Implementation practices](#appendix-implementation-practices).
+
+## Connector API
+
+Connectors MUST provide a service with their application ID, that is, the same name as the name that they sent their `org.unifiedpush.Distributor1.Register` call from.
+
+Connectors MUST implement the `org.unifiedpush.Connector1` interface at the object path `/org/unifiedpush/Connector`. The caller of the methods in this interface MUST NOT wait for a response from them!
+
+### org.unifiedpush.Connector1.Message (String, String, String) → nothing
+
+The distributor calls this method to send a new push message to the connector.
+
+The first argument is the token of the connection, the second argument is the raw POST data as received by the distributor, and the third argument is optionally an ID identifying the message (otherwise it is an empty string). This method does not return anything.
+
+### org.unifiedpush.Connector1.NewEndpoint (String, String) → nothing
+
+The distributor calls this method to inform the connector of the endpoint URL, both after registering for the first time, and if the endpoint changes afterwards. In addition to implementing this method, the connector SHOULD also call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) on every startup to request the newest endpoint URL, in case it missed this call otherwise.
+
+The first argument is the token of the connection, and the second argument is the new endpoint. This method does not return anything.
+
+### org.unifiedpush.Connector1.Unregister (String) → nothing
+
+The distributor calls this method to inform the connector that it has been unregistered for some server-side or distributor-side reason. If the app wants to remain registered, it SHOULD simply call [org.unifiedpush.Distributor1.Register](#orgunifiedpushdistributor1register-string--string-string) again after getting unregistered. A response indicating "REGISTRATION_REFUSED" would mean that the app is no longer allowed to register.
+
+The argument of the method is the token of the connection. This method does not return anything.
```

Comment:
The distributor MUST call this method to the registered application to confirm unregistration or to inform the application about unregistration.

If this action is send to inform the application, the intent MUST have the following argument:

* token: the token supplied by the end user application during registration


Other part is implementation or somehting that isn't related to Unregistered